doc/source/train/user-guides/data-loading-preprocessing.rst

```diff
@@ -44,18 +49,21 @@ Using Ray Data and Ray Train for distributed training on large datasets involves
             # Step 1: Create a Ray Dataset from in-memory Python lists.
             # You can also create a Ray Dataset from many other sources and file
             # formats.
-            train_dataset = ray.data.from_items([{"x": x, "y": 2 * x + 1} for x in range(200)])
+            train_dataset = ray.data.from_items([{"x": x, "y": 2 * x} for x in range(200)])
+
+            # Step 2: Preprocess your Ray Dataset.
+            def increment(batch):
+                batch["y"] = batch["y"] + 1
+                return batch
+
+            train_dataset = train_dataset.map_batches(increment)
 
-            # Step 2: Define your training function. This function contains the logic
-            # for creating the model and the training loop to train the model.
-            # See the Ray Train user guides for information such as how to report
-            # metrics or periodically save model checkpoints.
             def train_func(config):
                 model = nn.Sequential(nn.Linear(1, 1), nn.Sigmoid())
                 loss_fn = torch.nn.BCELoss()
                 optimizer = torch.optim.SGD(model.parameters(), lr=0.001)
 
-                # Step 3: Access the dataset shard for the training worker via
+                # Step 4: Access the dataset shard for the training worker via
```

Comment:
I am thinking if we can simplify this `train_func`, it's kind of hard to follow now. Since we are discussing data loading here, we probably don't need to show how to train a model and report a checkpoint.

```python
def train_func()
    # Step x: ...
    train_ds = train.get_dataset_shard("train") # Sharded
    val_ds = train.get_dataset_shard("val") # Unsharded

    # Step y: ...
    for batch in ds.iter_torch_batches(batch_size=16):
        assert type(batch["x"]) == torch.Tensor
        assert type(batch["y"]) == torch.Tensor
        assert batch["x"].shape == (16,)
        assert batch["y"].shape == (16,)
```

Having some code like above should be enough?